src/mlpack/methods/hmm/hmm_model.hpp

```diff
@@ -129,6 +129,20 @@ class HMMModel
     return *this;
   }
 
+  //! Move assignment operator.
+  HMMModel& operator=(HMMModel&& other)
+  {
+    if (this != &other)
+    {
+      type = other.type;
+      discreteHMM = other.discreteHMM;
+      gaussianHMM = other.gaussianHMM;
+      gmmHMM = other.gmmHMM;
+      diagGMMHMM = other.diagGMMHMM;
```

Comment:
Do we need to clear the members of `other`?

Reply:
Yes, we do! Thanks for that!